cloudinit/config/cc_ubuntu_advantage.py

```diff
@@ -246,7 +248,36 @@ def maybe_install_ua_tools(cloud):
         raise
 
 
-def handle(name, cfg, cloud, log, args):
+class NotAProInstance(Exception):
```

Comment:
We agreed to use the `uaclient.api.u.pro.detect.should_auto_attach.v1 import should_auto_attach_on_machine` new API function to check if we are in a Pro instance.

Should we also catch `NonAutoAttachImageError` in the call to `with_short_retry`?

@CalvoM

Reply:
Yes you could catch it to be thorough, but since `should_auto_attach` checks for it as well, it might not be necessary.
`should_auto_attach` endpoint performs an extra check which is checking for the installation of `ua pro` which cannot be done in `full_auto_attach` (updated name for `with_short_retry`). Furthermore, we do the checking in `full_auto_attach` as side effect of trying to get the cloud instance which we use later on in the function.